crates/oxc_linter/src/rules/eslint/sort_keys.rs

```diff
@@ -0,0 +1,1093 @@
+use crate::{context::LintContext, rule::Rule, AstNode};
+use itertools::all;
+use oxc_ast::ast::ObjectPropertyKind;
+use oxc_ast::syntax_directed_operations::PropName;
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+use std::cmp::Ordering;
+use std::str::Chars;
+
+#[derive(Debug, Default, Clone)]
+pub struct SortKeys(Box<SortKeysOptions>);
+
+#[derive(Debug, Default, Clone, Eq, PartialEq)]
+pub enum SortOrder {
+    Desc,
+    #[default]
+    Asc,
+}
+
+#[derive(Debug, Clone)]
+pub struct SortKeysOptions {
+    sort_order: SortOrder,
+    case_sensitive: bool,
+    natural: bool,
+    min_keys: usize,
+    allow_line_separated_groups: bool,
+}
+
+impl Default for SortKeysOptions {
+    fn default() -> Self {
+        // we follow the eslint defaults
+        Self {
+            sort_order: SortOrder::Asc,
+            case_sensitive: true,
+            natural: false,
+            min_keys: 2,
+            allow_line_separated_groups: false,
+        }
+    }
+}
+
+impl std::ops::Deref for SortKeys {
+    type Target = SortKeysOptions;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+fn sort_properties_diagnostic(span0: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Object keys should be sorted").with_label(span0)
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// When declaring multiple properties, sorting property names alphabetically makes it easier
+    /// to find and/or diff necessary properties at a later time.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Unsorted property keys can make the code harder to read and maintain.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// let myObj = {
+    ///   c: 1,
+    ///   a: 2,
+    /// };
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// let myObj = {
+    ///   a: 2,
+    ///   c: 1,
+    /// };
+    /// ```
+    SortKeys,
+    pedantic, // TODO: change category to `correctness`, `suspicious`, `pedantic`, `perf`, `restriction`, or `style`
+             // See <https://oxc.rs/docs/contribute/linter.html#rule-category> for details
+
+    pending  // TODO: describe fix capabilities. Remove if no fix can be done,
+             // keep at 'pending' if you think one could be added but don't know how.
+             // Options are 'fix', 'fix_dangerous', 'suggestion', and 'conditional_fix_suggestion'
```

Comment:
The TODO comments can be removed as they placeholder has already been replaced.

Reply:
fixed